src/main/java/com/lambdaschool/oktafoundation/controllers/ReportController.java

```diff
@@ -0,0 +1,205 @@
+        emojimap.put("1F601", 2);
+        emojimap.put("1F642", 1);
+        emojimap.put("1F610", 0);
+        emojimap.put("1F641", -1);
+        emojimap.put("1F61E", -2);
+    }
+
+    @Autowired
+    private MemberReactionService memberReactionService;
+
+    @Autowired
+    private ClubRepository clubRepository;
+
+    @Autowired
+    private EntityManager entityManager;
+
+    /* The endpoints shall serve data analysis of the member reactions
+    with respect to club/activity/member.
+     */
+
+    // Currently we have the simple feature of calculating the avgs of positivity value of all members
+    // and avgs of the received posivity regarding activities in a club.
+
+
+    @PreAuthorize("hasAnyRole('ADMIN','CD')")
+    @GetMapping(value = "/club/{cid}/members")
+    public ResponseEntity<?> getPosivitiesByMembers(
+            @PathVariable Long cid,
+            @RequestParam(value = "from", required = false) String from,
+            @RequestParam(value = "to", required = false) String to){
+
+        var fromdate = from;
+        var todate = to;
+        if(from == null){
+            fromdate = "1000-01-01";
+        }
+        if(to == null){
+            todate = "3000-01-01";
+        }
+
+        // whether we guard against certain club
+        // if cid = 0; we search all members
+        var clubfilter = "";
+        if (cid.intValue() != 0){
+            clubfilter = "and clubid = " + cid.toString()+ " ";
+        }
+
+
+        var q = entityManager.createNativeQuery("select * from memberreactions where created_date >= date'" + fromdate + "'" +" and created_date <= date'" + todate + "'"  + clubfilter,MemberReactions.class);
+        List<MemberReactions> mrlist = q.getResultList();
+        var prememberPositivty = new HashMap<String, ArrayList<Integer>>();
+//        System.out.println(mrlist.get(0));
+        mrlist.forEach(e -> {
+            if(prememberPositivty.containsKey(e.getMember().getMemberid())){
+                prememberPositivty.get(e.getMember().getMemberid()).add(emojimap.get(e.getReaction().getReactionvalue()));
+            } else {
+                var templ = new ArrayList<Integer> ();
+                templ.add(emojimap.get(e.getReaction().getReactionvalue()));
+                prememberPositivty.put(e.getMember().getMemberid(),templ);
+            }
+        });
+
+        List<MemberPosivity> mplist = new ArrayList<>();
+        var members = new ArrayList<>(prememberPositivty.keySet());
+        for (var member: members) {
+            var temp = new MemberPosivity();
+            temp.setMemberid(member);
+            var memberreactions = prememberPositivty.get(member);
+            var len = memberreactions.size();
+            temp.setPosivitiy(memberreactions.stream().reduce((a,b) -> a+b).get()/(double)len);
+            mplist.add(temp);
+        }
+
+
+        return new ResponseEntity<>(mplist,HttpStatus.OK);
+    }
+
+
+    @PreAuthorize("hasAnyRole('ADMIN','CD')")
```

Comment:
return an activity avg emovalue against all its members for all activities.
If cid = 0, we run it against all clubs, otherwise, restrict to a single club